lib/distributed_cache.rb

```diff
@@ -3,6 +3,26 @@
 require 'message_bus/distributed_cache'
 
 class DistributedCache < MessageBus::DistributedCache
+  class << self
+    if Rails.env.test?
+      def caches
+        @caches ||= {}
+      end
+
+      def get(key, opts = {})
+        caches[key] ||= new(key, opts)
+      end
+
+      def clear_caches!
+        caches.each_value do |cache|
```

Comment:
what is the perf impact of adding this to all tests?

Reply:
~40 seconds.

Reply:
which is about ~5% for me.

Reply:
Would `@caches = {}` not do it?

Reply:
@eviltrout, unfortunately not, references to these caches are held in instances variables on classes themselves.

Reply:
Maybe we could DRY up the DistributedCache memoizing. We could create a mixin `HasDistributedCache` that classes would add, then they'd use a class method `distributed_cache :last_read_only` which creates the memoized cache.

In test mode we'd make it something we could easily clean up. Otherwise we'd memoize the instance.

Reply:
That would work, I'll do that.